Sources/TuistGenerator/Linter/TargetLinter.swift

```diff
@@ -139,4 +140,28 @@ class TargetLinter: TargetLinting {
 
         return []
     }
+
+    private func lintDeploymentTarget(target: Target) -> [LintingIssue] {
+        guard let deploymentTarget = target.deploymentTarget else {
+            return []
+        }
+
+        let issue = LintingIssue(reason: "The version of deployment target is incorrect", severity: .error)
+
+        let osVersionRegex = "\\b[0-9]+\\.[0-9]+(?:\\.[0-9]+)?\\b"
+        switch deploymentTarget {
```

Comment:
What about extending the type of this variable to expose a `.platform` attribute? That way you don't need a switch here:

```swift
let platform = deploymentTarget.platform
if !version.matches(pattern: osVersionRegex) { return [issue] }
return []
```

Reply:
Hmm, you meant `version` attribute? 

```
let version = deploymentTarget.version
if !version.matches(pattern: osVersionRegex) { return [issue] }
return []
```